server/odc-service/src/main/java/com/oceanbase/odc/service/notification/helper/EventBuilder.java

```diff
@@ -321,6 +331,34 @@ private <T> void resolveLabels(EventLabels labels, T task) {
                 log.warn("failed to query project info.", e);
             }
         }
+        try {
+            labels.putIfNonNull(TICKET_URL, getTicketUrl(labels, task));
+        } catch (Exception e) {
+            log.warn("failed to get ticket url.", e);
+        }
+    }
+
+    private <T> String getTicketUrl(EventLabels labels, T task) {
+        String taskType = labels.get(TASK_TYPE);
+        String taskId = labels.get(TASK_ID);
+        Long organizationId;
+        if (task instanceof TaskEntity) {
+            organizationId = ((TaskEntity) task).getOrganizationId();
+        } else {
+            organizationId = ((ScheduleEntity) task).getOrganizationId();
+        }
+
+        String host = Optional.ofNullable(hostProperties.getOdcHost()).orElse(SystemUtils.getLocalIpAddress());
```

Comment:
1. lets' extract siteUrl cacluation logic to a separated component for reuse.

2. systemConfigService.queryByKeyPrefix, SystemConfigService may add a new method for queryByKey.

Reply:
I'll do these in the next PR~